include/cppkafka/utils/roundrobin_poll_adapter.h

```diff
@@ -173,25 +174,39 @@ class RoundRobinPollAdapter
      */
     MessageList poll_batch(size_t max_batch_size, std::chrono::milliseconds timeout);
     
+    /**
+     * \brief Gets the number of assigned partitions that can be polled across all topics
+     *
+     * \return The number of partitions
+     */
+    size_t get_num_partitions();
+    
 private:
     class CircularBuffer {
-        using qlist = std::list<Queue>;
-        using qiter = qlist::iterator;
     public:
-        qlist& ref() { return queues_; }
+        // typedefs
+        using toppar_t = std::pair<std::string, int>; //<topic, partition>
+        using qmap_t = std::map<toppar_t, Queue>;
+        using qiter_t = qmap_t::iterator;
```

Comment:
Is this type alias really necessary? The type is only used once and only internally.